synapseclient/core/credentials/credential_provider.py

```diff
@@ -104,48 +98,36 @@ def _create_synapse_credential(
 
 
 class UserArgsCredentialsProvider(SynapseCredentialsProvider):
+    def _get_auth_info(self, syn, user_login_args):
+
+        username = user_login_args.get("username")
+        token = user_login_args.get("auth_token")
+
+        if username and token:
+            return username, token
+
+        return None, None
     """
     Retrieves auth info from user_login_args during a CLI session.
     """
 
-    def _get_auth_info(
-        self, syn: "Synapse", user_login_args: Dict[str, str]
-    ) -> Tuple[str, str]:
-        return (
-            user_login_args.username,
-            user_login_args.auth_token,
-        )
-
-
```

Comment:
If we change `UserLoginArgs` to be a dataclass there are some changes needed needed throughout this `credential_provider.py` script. Since everything would either be an instance of `UserLoginArgs`, or waiting to be converted into an instance of the `UserLoginArgs` dataclass.